lib/model/query/submission-attachments.js

```diff
@@ -100,21 +100,48 @@ const create = (submission, form, binaryFields, files = [], deprecated = []) =>
   });
 };
 
-const upsert = (def, files) => ({ Blobs, SubmissionAttachments }) =>
+const upsert = (submission, form, def, files) => ({ run, Blobs, SubmissionAttachments, context }) =>
   SubmissionAttachments.getAllByDefId(def.id)
     .then((expecteds) => {
       const lookup = new Set(expecteds.map((att) => att.name));
       const present = files.filter((file) => lookup.has(file.fieldname));
-      return Promise.all(present
-        .map((file) => Blobs.ensure(Blob.fromBuffer(file.buffer, file.mimetype || defaultMimetypeFor(file.fieldname)))
-          .then((blobId) => SubmissionAttachments.attach(def.id, file.fieldname, blobId))));
+      return Promise.all(
+        present.map((file) => Blobs.ensure(Blob.fromBuffer(file.buffer, file.mimetype || defaultMimetypeFor(file.fieldname)))
+          .then((blobId) => SubmissionAttachments.attach(def.id, file.fieldname, blobId))
+        )
+      ).then(linkResults =>
+        // auditlog the updates (only those which were added or of which the content changed)
+        linkResults
+          .filter(el => el.blobBefore !== el.blobNow)
```

Comment:
If you keep the query in `attach()` the same, then blobs that you tried to attach but were already set will have a null `blobNow`, which you could then skip in the logging. Or you could tweak the query. 

```suggestion
          .filter(el => el.blobNow && el.blobBefore !== el.blobNow)
```